File: Problem 1 csharp/Part1.cs
// See https://aka.ms/new-console-template for more information
/* PART 1
var sr = new StreamReader("TestData.txt");
string? s = "";
List<int> values = new List<int>();
int sum = 0;
while ((s = sr.ReadLine()) != null)
{
    if(s == "")
    {
        values.Add(sum);
        sum = 0;
    }
    else
    {
        sum += int.Parse(s);
    }
}
if(sum != 0)
{
    values.Add(sum);
}
int max = 0;

foreach(int i in values)
{
    if(i > max)
    {
        max = i;
    }
}

Console.WriteLine(max);
*/

File: Problem1_Rust/src/part1.rs
//This solves only Part 1

use std::fs;

fn main() {
    let data = fs::read_to_string("TestData.txt");
    let binding = data.unwrap();
    let lines = binding.lines();
    let mut sum = 0;
    let mut vector = Vec::<i32>::new();
    for line in lines {
        if line == "" {
            vector.push(sum);
            sum = 0;
        } else {
            let num = line.parse::<i32>().unwrap();
            sum += num;
        }
    }
    let mut max = 0;
    for i in vector {
        if i > max {
            max = i;
        }
    }
    println!("{}", max);
}